test/Interop/Cxx/class/initializer-typechecker.swift

```diff
@@ -0,0 +1,15 @@
+// RUN: %target-typecheck-verify-swift -I %S/Inputs -enable-cxx-interop
+
+import Initializer
+
+let structPrivateOnly = StructPrivateOnly(x: 42)  // expected-error {{argument passed to call that takes no arguments}}
+let structPublicOnly = StructPublicOnly(x: 42)
+let structEmptyPrivateSetion = StructEmptyPrivateSection(x: 42)
+let structPublicAndPrivate1 = StructPublicAndPrivate(x: 42)  // expected-error {{argument passed to call that takes no arguments}}
+let structPublicAndPrivate2 = StructPublicAndPrivate(x: 42, y: 23)  // expected-error {{argument passed to call that takes no arguments}}
+
+let classPrivateOnly = ClassPrivateOnly(x: 42) // expected-error {{argument passed to call that takes no arguments}}
+let classPublicOnly = ClassPublicOnly(x: 42)
+let classEmptyPublicSetion = ClassEmptyPublicSection(x: 42) // expected-error {{argument passed to call that takes no arguments}}
+let classPublicAndPrivate1 = ClassPrivateAndPublic(y: 23)  // expected-error {{argument passed to call that takes no arguments}}
+let classPublicAndPrivate2 = ClassPrivateAndPublic(x: 42, y: 23)  // expected-error {{argument passed to call that takes no arguments}}
```

Comment:
Thanks for the thorough test! I have two suggestions:

- make it more obvious in variable names which are public and which are private (`varPublic`, `varPrivate` etc.) so that it is easy to read the Swift file without having to cross-reference the header.

- consider adding a module interface printing test -- it would clearly show what API is imported. If you prefer, you could even drop the type checker test at that point -- but more tests are always welcome.

Reply:
Done (both).

Do we consider the module interface tests to be authoritative enough? I feel like those don't fully test that actual interaction with the imported API. I guess now that I have the type checker test I'd like to keep it.

Reply:
You're right, module interface printing tests are best effort.